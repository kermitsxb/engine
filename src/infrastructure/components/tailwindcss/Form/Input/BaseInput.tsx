type InputLabelProps = {
  label: string
  field: string
  required?: boolean
}

export const InputLabel = ({ label, field, required = false }: InputLabelProps) => {
  return (
    <label htmlFor={field} className="block mb-2 text-sm text-gray-700 font-medium dark:text-white">
      {label}
      {required ? <span className="text-red-500"> *</span> : null}
    </label>
  )
}

type InputDescriptionProps = {
  description: string
}

export const InputDescription = ({ description }: InputDescriptionProps) => {
  return <p className="mb-2 text-sm text-gray-500 dark:text-neutral-500">{description}</p>
}

type InputContextProps = {
  label?: string
  description?: string
  field: string
  required?: boolean
  children: React.ReactNode
}

export const InputContext = ({
  label,
  description,
  field,
  required,
  children,
}: InputContextProps) => {
  return (
    <div>
      {label ? <InputLabel label={label} field={field} required={required} /> : null}
      {description ? <InputDescription description={description} /> : null}
      {children}
    </div>
  )
}
