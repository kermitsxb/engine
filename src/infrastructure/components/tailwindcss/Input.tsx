import type { InputProps } from '/domain/entities/Form/Input'

export const Input = ({ field, type, label, description, placeholder, required }: InputProps) => {
  return (
    <div>
      {label ? (
        <label
          htmlFor={field}
          className="block mb-2 text-sm text-gray-700 font-medium dark:text-white"
        >
          {label}
        </label>
      ) : null}
      {description ? (
        <p
          className="mb-2 text-sm text-gray-500 dark:text-neutral-500"
          id={'description-' + description}
        >
          {description}
        </p>
      ) : null}
      <input
        type={type}
        name={field}
        placeholder={placeholder}
        required={!!required}
        className="py-2.5 sm:py-3 px-4 block w-full bg-white border border-gray-200 rounded-lg sm:text-sm focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600"
        aria-describedby={description ? 'description-' + field : undefined}
      />
    </div>
  )
}
