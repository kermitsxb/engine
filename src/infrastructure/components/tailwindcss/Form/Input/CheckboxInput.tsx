import { InputContext } from './BaseInput'
import type { CheckboxInputProps } from '/domain/components/Form/Input/CheckboxInput'

export const CheckboxInput = ({ field, label, description, required }: CheckboxInputProps) => {
  return (
    <InputContext label={label} description={description} field={field} required={required}>
      <input
        type="checkbox"
        name={field}
        required={!!required}
        className="shrink-0 mt-0.5 border-gray-200 rounded-sm text-blue-600 focus:ring-blue-500 checked:border-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-800 dark:border-neutral-700 dark:checked:bg-blue-500 dark:checked:border-blue-500 dark:focus:ring-offset-gray-800"
      />
    </InputContext>
  )
}
