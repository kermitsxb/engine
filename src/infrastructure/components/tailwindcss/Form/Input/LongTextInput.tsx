import { InputContext } from './BaseInput'
import type { LongTextInputProps } from '/domain/components/Form/Input/LongTextInput'

export const LongTextInput = ({
  field,
  label,
  description,
  placeholder,
  required,
}: LongTextInputProps) => {
  return (
    <InputContext label={label} description={description} field={field} required={required}>
      <textarea
        id={field}
        name={field}
        required={!!required}
        className="block w-full rounded-lg border-gray-200 px-3 py-2 focus:border-blue-500 focus:ring-blue-500 disabled:pointer-events-none disabled:opacity-50 sm:px-4 sm:py-3 sm:text-sm dark:border-neutral-700 dark:bg-neutral-900 dark:text-neutral-400 dark:placeholder-neutral-500 dark:focus:ring-neutral-600"
        rows={3}
        placeholder={placeholder}
        data-hs-textarea-auto-height=""
      ></textarea>
    </InputContext>
  )
}
