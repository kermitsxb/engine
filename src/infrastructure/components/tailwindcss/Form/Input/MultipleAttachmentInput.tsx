import { InputContext } from './BaseInput'
import type { MultipleAttachmentInputProps } from '/domain/components/Form/Input/MultipleAttachmentInput'

// https://preline.co/docs/file-input.html

export const MultipleAttachmentInput = ({
  field,
  label,
  description,
  placeholder,
  required,
}: MultipleAttachmentInputProps) => {
  return (
    <InputContext label={label} description={description} field={field} required={required}>
      <label className="block">
        <span className="sr-only">{placeholder}</span>
        <input
          type="file"
          name={field}
          multiple
          required={!!required}
          className="block w-full text-sm text-gray-500
            file:me-4 file:py-2 file:px-4
            file:rounded-lg file:border-0
            file:text-sm file:font-semibold
            file:bg-blue-600 file:text-white
            hover:file:bg-blue-700
            file:disabled:opacity-50 file:disabled:pointer-events-none
            dark:text-neutral-500
            dark:file:bg-blue-500
            dark:hover:file:bg-blue-400
          "
        />
      </label>
    </InputContext>
  )
}
