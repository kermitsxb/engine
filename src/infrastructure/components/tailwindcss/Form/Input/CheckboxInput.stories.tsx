import type { Meta, StoryObj } from '@storybook/react'

import { CheckboxInput } from './CheckboxInput'
import { Form } from '../../Form/Form'
import type { CheckboxInputProps } from '/domain/components/Form/Input/CheckboxInput'

const CheckboxInputWithForm = (props: CheckboxInputProps) => (
  <Form id="id" title="Form" submitLabel="Save">
    <CheckboxInput {...props} />
  </Form>
)

const meta = {
  title: 'Form/CheckboxInput',
  component: CheckboxInputWithForm,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof CheckboxInputWithForm>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    field: 'checkbox',
    label: 'Checkbox',
    description: 'This is a description',
  },
}
