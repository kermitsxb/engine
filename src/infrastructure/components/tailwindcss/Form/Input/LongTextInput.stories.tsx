import type { Meta, StoryObj } from '@storybook/react'

import { LongTextInput } from './LongTextInput'
import { Form } from '../Form'
import type { LongTextInputProps } from '/domain/components/Form/Input/LongTextInput'

const LongTextInputWithForm = (props: LongTextInputProps) => (
  <Form id="id" title="Form" submitLabel="Save">
    <LongTextInput {...props} />
  </Form>
)

const meta = {
  title: 'Form/LongTextInput',
  component: LongTextInputWithForm,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof LongTextInputWithForm>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    field: 'single_line_text',
    label: 'Single Line Text',
    description: 'This is a description',
    placeholder: 'Enter your text',
  },
}

export const Required: Story = {
  args: {
    ...Default.args,
    required: true,
  },
}
