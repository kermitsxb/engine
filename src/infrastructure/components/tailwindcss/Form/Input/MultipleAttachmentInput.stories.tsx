import type { Meta, StoryObj } from '@storybook/react'

import { MultipleAttachmentInput } from './MultipleAttachmentInput'
import { Form } from '../Form'
import type { MultipleAttachmentInputProps } from '/domain/components/Form/Input/MultipleAttachmentInput'

const MultipleAttachmentInputWithForm = (props: MultipleAttachmentInputProps) => (
  <Form id="id" title="Form" submitLabel="Save">
    <MultipleAttachmentInput {...props} />
  </Form>
)

const meta = {
  title: 'Form/MultipleAttachmentInput',
  component: MultipleAttachmentInputWithForm,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof MultipleAttachmentInputWithForm>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    field: 'multiple_attachment',
    label: 'Multiple Attachment',
    description: 'This is a description',
    placeholder: 'Select files',
  },
}

export const Required: Story = {
  args: {
    ...Default.args,
    required: true,
  },
}
