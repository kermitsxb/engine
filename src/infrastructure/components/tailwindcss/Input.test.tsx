import { describe, it, expect } from 'bun:test'
import { Input } from './Input'
import { renderToString } from 'react-dom/server'

describe('Input', () => {
  it('should match the form snapshot', async () => {
    // GIVEN
    const input = <Input field="name" type="text" />

    // WHEN
    const html = renderToString(input)

    // THEN
    expect(html).toMatchSnapshot()
  })

  it('should match the input snapshot with label', async () => {
    // GIVEN
    const input = <Input field="name" type="text" label="Name" />

    // WHEN
    const html = renderToString(input)

    // THEN
    expect(html).toMatchSnapshot()
  })

  it('should match the input snapshot with description', async () => {
    // GIVEN
    const input = <Input field="name" type="text" description="Description" />

    // WHEN
    const html = renderToString(input)

    // THEN
    expect(html).toMatchSnapshot()
  })

  it('should match the input snapshot with label and description', async () => {
    // GIVEN
    const input = <Input field="name" type="text" label="Name" description="Description" />

    // WHEN
    const html = renderToString(input)

    // THEN
    expect(html).toMatchSnapshot()
  })

  it('should match the input snapshot with placeholder', async () => {
    // GIVEN
    const input = <Input field="name" type="text" placeholder="Placeholder" />

    // WHEN
    const html = renderToString(input)

    // THEN
    expect(html).toMatchSnapshot()
  })

  it('should match the input snapshot with required', async () => {
    // GIVEN
    const input = <Input field="name" type="text" required />

    // WHEN
    const html = renderToString(input)

    // THEN
    expect(html).toMatchSnapshot()
  })
})
