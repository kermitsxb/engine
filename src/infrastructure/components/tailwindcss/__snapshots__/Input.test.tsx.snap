// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match the form snapshot 1`] = `"<div><input type="text" class="py-2.5 sm:py-3 px-4 block w-full bg-white border border-gray-200 rounded-lg sm:text-sm focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="name"/></div>"`;

exports[`Input should match the input snapshot with label 1`] = `"<div><label for="name" class="block mb-2 text-sm text-gray-700 font-medium dark:text-white">Name</label><input type="text" class="py-2.5 sm:py-3 px-4 block w-full bg-white border border-gray-200 rounded-lg sm:text-sm focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="name"/></div>"`;

exports[`Input should match the input snapshot with description 1`] = `"<div><p class="mb-2 text-sm text-gray-500 dark:text-neutral-500" id="description-Description">Description</p><input type="text" class="py-2.5 sm:py-3 px-4 block w-full bg-white border border-gray-200 rounded-lg sm:text-sm focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" aria-describedby="description-name" name="name"/></div>"`;

exports[`Input should match the input snapshot with label and description 1`] = `"<div><label for="name" class="block mb-2 text-sm text-gray-700 font-medium dark:text-white">Name</label><p class="mb-2 text-sm text-gray-500 dark:text-neutral-500" id="description-Description">Description</p><input type="text" class="py-2.5 sm:py-3 px-4 block w-full bg-white border border-gray-200 rounded-lg sm:text-sm focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" aria-describedby="description-name" name="name"/></div>"`;

exports[`Input should match the input snapshot with placeholder 1`] = `"<div><input type="text" placeholder="Placeholder" class="py-2.5 sm:py-3 px-4 block w-full bg-white border border-gray-200 rounded-lg sm:text-sm focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="name"/></div>"`;

exports[`Input should match the input snapshot with required 1`] = `"<div><input type="text" required="" class="py-2.5 sm:py-3 px-4 block w-full bg-white border border-gray-200 rounded-lg sm:text-sm focus:border-blue-500 focus:ring-blue-500 disabled:opacity-50 disabled:pointer-events-none dark:bg-neutral-900 dark:border-neutral-700 dark:text-white dark:placeholder-neutral-500 dark:focus:ring-neutral-600" name="name"/></div>"`;
